Modules/modFileEncoding.ACM

```diff
@@ -0,0 +1,257 @@
+Option Compare Database
+Option Explicit
+
+Private Declare Sub CopyMemory Lib "kernel32" Alias "RtlMoveMemory" (ByVal Destination As Any, ByVal Source As Any, ByVal Length As Long)
+
+Function IsUCS2(strFilename As String) As Boolean
+  ' Returns true if the file specified is in UCS2 little endian format
+  Dim f As Long
+  Dim byts() As Byte
+  
+  If FileExists(strFilename) Then
+    ' load the file into the bytsFile array
+    f = FreeFile()
+    Open strFilename For Binary Access Read As #f
+    ReDim byts(2)
+    Get #f, 1, byts()
+    Close #f
+    IsUCS2 = ((byts(0) = 255) And (byts(1) = 254))
+  End If
+End Function
+
+Function ConvFileUTF8_to_UCS2(strFilename As String, Optional strNewFilename As String) As Boolean
+  ' Converts the supplied file from UTF8 to UCS2 format
+  Dim byts() As Byte
+  Dim bytsOut() As Byte
+  
+  If LenB(strNewFilename) = 0 Then strNewFilename = strFilename
+  If ReadFromFile(strFilename, byts) Then
+    If UTF8_to_UCS2(byts, bytsOut) Then
+      ConvFileUTF8_to_UCS2 = WriteToFile(bytsOut, strNewFilename)
+    End If
+  End If
+    
+End Function
+
+Function ConvFileUCS2_to_UTF8(strFilename As String, Optional strNewFilename As String) As Boolean
+  ' Converts the supplied file from UCS2 to UTF8 format
+  Dim byts() As Byte
+  Dim bytsOut() As Byte
+  
+  If LenB(strNewFilename) = 0 Then strNewFilename = strFilename
+  If ReadFromFile(strFilename, byts) Then
+    If UCS2_to_UTF8(byts, bytsOut) Then

```

Comment:
If this fails we should warn the user. And possibly delete the file or put the old file back?

Reply:
The calling function knows whether it succeeded or not. If it fails the original file will still be there because the file is not overwritten until the very last call. Calling functions modified to warn user.